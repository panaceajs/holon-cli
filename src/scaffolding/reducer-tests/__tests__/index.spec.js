const { resolve } = require('path');
const snapshot = require('../../../utils/task-snapshot');
const task = require('../');

describe('action-types', () => {
  it('should create action types in `action-types/index.js`', async () => {
    await expect(
      snapshot(task, {
        create: true,
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });

  it('should create action types in `other-dir/action-types/index.js`', async () => {
    await expect(
      snapshot(task, {
        create: true,
        cwd: resolve('other-dir'),
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });

  it('should update action types in `./action-types/index.js`', async () => {
    await expect(
      snapshot(task, {
        cwd: resolve(__dirname, '__fixtures__'),
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });
});
