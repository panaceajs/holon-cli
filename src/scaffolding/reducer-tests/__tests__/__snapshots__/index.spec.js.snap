// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action-types should create action types in \`action-types/index.js\` 1`] = `
"
--------------------------------------------------------------------------------
// ./action-types/index.js
// Create
--------------------------------------------------------------------------------

export const ANOTHER_ACTION = 'ANOTHER_ACTION';
export const AN_ACTION = 'AN_ACTION';



"
`;

exports[`action-types should create action types in \`other-dir/action-types/index.js\` 1`] = `
"
--------------------------------------------------------------------------------
// ./action-types/index.js
// Create
--------------------------------------------------------------------------------

export const ANOTHER_ACTION = 'ANOTHER_ACTION';
export const AN_ACTION = 'AN_ACTION';



"
`;

exports[`action-types should update action types in \`./action-types/index.js\` 1`] = `
"
--------------------------------------------------------------------------------
// ./action-types/index.js
// Update
--------------------------------------------------------------------------------

export const EXISTING_ACTION = 'EXISTING_ACTION';
export const ANOTHER_ACTION =  'ANOTHER_ACTION_WITH_A_VALUE_WHICH_SHOULD_NOT_BE_OVERWRITTEN';

      ↓ ↓ ↓ ↓ ↓ ↓



export const EXISTING_ACTION = 'EXISTING_ACTION';
export const ANOTHER_ACTION = 'ANOTHER_ACTION_WITH_A_VALUE_WHICH_SHOULD_NOT_BE_OVERWRITTEN';
export const AN_ACTION = \\"AN_ACTION\\";


"
`;
